### DISCUSSION TITLE
Node edge layout issue/questions

### DISCUSSION THREAD

**Discussion by quickfold** [freeplane/freeplane#287] [Help]
1. I have noticed that the edges connecting nodes have a different layout if the root is selected vs. another node. See first two images. Is this expected or a bug?
2. I see that in some cases that edges cross each other. Changing the length of text in the nodes can make these crossovers appear or disappear, but not in a way I can make sense of. See second two images. This line crossing is bothersome to me, although I grant it isn't a real functionality issue. Is this a known issue? Thanks.

![image](https://user-images.githubusercontent.com/2906323/160949250-ddd95187-3dda-4956-9168-3e4865a2524a.png)
![image](https://user-images.githubusercontent.com/2906323/160949275-1ae96fd7-a5bc-421e-91f8-1c6aa8ff8992.png)
![image](https://user-images.githubusercontent.com/2906323/160949298-5dcc77f6-5515-42a7-b4b5-273b7045586a.png)
![image](https://user-images.githubusercontent.com/2906323/160949313-e1b0f445-5b73-48ce-a9bf-23a65626c114.png)

**Comment by rexelbartolome** [comment]
I think it's because it's switching between the `Selected` style and a default `Level 1` style where the `Selected` has a **Bubble** node shape and the `Level 1` style has a **Fork** node shape

https://user-images.githubusercontent.com/88552647/160950411-12dd9daa-6059-431c-bffc-4669d23ab98c.mp4

The lines crossing over I don't really know, but it maybe related to how FP assumes all of these nodes have the same node shape and changing one of them to be different causes a different layout to be done? 🤷‍♂️

**Comment by quickfold** [comment]
I agree, the node shape seems to affect the edge layout. Not sure if this is a bug or intended, or if the edge crossing is due to this. Dimitry, any comment? @dpolivaev

**Reply by dpolivaev** [reply]
I could have a look at it if you attached your example as a mind map file, not just as a picture.
In my eyes guessing why it behaves as it does without the map is impossible.
